packages/mdc-textfield/foundation.ts

```diff
@@ -246,17 +229,12 @@ class MDCTextFieldFoundation extends MDCFoundation {
   /**
    * Sets the line ripple's transform origin, so that the line ripple activate
    * animation will animate out from the user's click location.
-   * @param {!Event} evt
    */
-  setTransformOrigin(evt) {
-    let targetEvent;
-    if (evt.touches) {
-      targetEvent = evt.touches[0];
-    } else {
-      targetEvent = evt;
-    }
-    const targetClientRect = targetEvent.target.getBoundingClientRect();
-    const normalizedX = targetEvent.clientX - targetClientRect.left;
+  setTransformOrigin(evt: Event): void {
```

Comment:
if you change the above types to SpecificEventListeners, then this won't require a cast to `as TouchEvent`

Reply:
We'll still need to cast it, because this param will be `evt: MouseEvent | TouchEvent`, and the two event types have different properties